src/modules/group/group.routes.ts

```diff
@@ -72,14 +74,162 @@ export const createGroupRoute = createRoute({
   },
 });
 
+export const getAllGroupsRoute = createRoute({
+  tags,
+  method: "get",
+  path: "/groups",
+  middleware: [
+    authMiddleware(),
+    requireAuth(),
+    checkRoleGuard(AuthRoles.ADMIN, AuthRoles.USER),
+  ] as const,
+  request: {
+    query: groupQuerySchema,
+  },
+  responses: {
+    [HTTPStatusCodes.OK]: jsonContent(
+      z.object({
+        success: z.boolean().default(true),
+        message: z.string(),
+        data: z.array(selectGroupSchema),
+      }),
+      "List of Groups received successfully",
+    ),
+    [HTTPStatusCodes.UNAUTHORIZED]: jsonContent(
+      z.object({
+        success: z.boolean().default(false),
+        message: z.string(),
+      }),
+      "You are not authorized, please login",
+    ),
+  },
+});
+
+export const getGroupById = createRoute({
+  tags,
+  method: "get",
+  path: "group/:id",
+  middleware: [
+    authMiddleware(),
+    requireAuth(),
+    checkRoleGuard(AuthRoles.ADMIN, AuthRoles.USER),
+  ] as const,
+  request: {
+    params: z.object({
+      id: z.string(),
```

Comment:
Add checks like minimum 32, maximum 60.

Let us also make a common id schema.

Reply:
Made common implementation check for min and max length of ID.